src/aklite_client_ext.cc

```diff
@@ -282,3 +283,45 @@ bool AkliteClientExt::RebootIfRequired() {
 
   return false;
 }
+
+InstallResult AkliteClientExt::Rollback(const LocalUpdateSource* local_update_source) {
+  auto current = GetCurrent();
+  auto pending_target = GetPendingTarget();
+  auto installation_in_progress = !pending_target.IsUnknown();
+  auto bad_target = installation_in_progress ? pending_target : current;
+
+  LOG_DEBUG << "User initiated rollback. Current Target is " << current.Name();
+  if (installation_in_progress) {
+    LOG_DEBUG << "There is an installation in progress of Target " << pending_target.Name();
+  }
+  auto storage = INvStorage::newStorage(client_->config.storage, false);
+  LOG_INFO << "Marking target " << bad_target.Name() << " as failed";
+  storage->saveInstalledVersion("", Target::fromTufTarget(bad_target), InstalledVersionUpdateMode::kBadTarget);
+
+  // Get rollback target
+  auto rollback_target = GetRollbackTarget(true, installation_in_progress);
```

Comment:
If this call takes place after reboot, then we shouldn't allow choosing the current target as the target to rollback to since the "bad" target is already current?
Or I missed something?